libs/ddhub-client-gateway-frontend/ui/channels/src/lib/containers/modals/Create/Restrictions/effects/didRestriction.effects.ts

```diff
@@ -14,7 +14,14 @@ export const useDIDRestrictionEffects = (currentDids: string[]) => {
 
   const didInputChangeHandler = (value: string) => {
     setDIDInput(value);
-    setIsValid(didRegex.test(value));
+  };
+
+  const didInputCheckValidity = () => {
```

Comment:
Are DID lengths always the same? Could we run this check in `onChange` as well, as long as `value` is the correct DID length? Only showing validation errors after moving away from the TextField might be a bit strange

Reply:
@nik-ewf yes, same length. I suppose that works too. We just want to prevent invalid DID error being displayed when user starts typing. I'll update the code.

Reply:
Yeah that makes sense, I saw where you were going with it 👍

Reply:
Another fun one you could look at is implementing a `debounce` function that checks after the user stops typing after ~300-500ms

Reply:
`/^did:[a-z0-9]+:([a-z0-9]+:)?(0x[0-9a-fA-F]{40})$/`

@nik-ewf The length 40 is only fixed for last pattern though. There's a minimum length for a DID to be valid according to the regex above (48 characters or above), so what I can do is add if statement in `onChange` in order to have error shown without having to click outside.

```
 setDIDInput(value);

    if (value.length >= 48) {
      setIsValid(didRegex.test(value));
    }
```

However, the code for `onBlur` still need to be there in case length is less than minimum or no error will be shown at all. I tried to add debounce as well, but the input field somehow freezes.

What do you think of this solution?

Reply:
Looks good to me, we could look at implementing debounce in the future 👍

Reply:
Implemented debounce in the new commit. Reference: https://www.carlrippon.com/using-lodash-debounce-with-react-and-ts/